crates/ruma-common/src/events/room/message/sanitize.rs

```diff
@@ -1,264 +1,87 @@
-use std::collections::{HashMap, HashSet};
+//! Convenience methods and types to sanitize text messages.
 
-use maplit::{hashmap, hashset};
-use scraper::Html;
+#[cfg(feature = "sanitize")]
+mod html_sanitizer;
 
-/// A sanitizer to filter [HTML tags and attributes] according to the Matrix specification.
+#[cfg(feature = "sanitize")]
+use html_sanitizer::HtmlSanitizer;
+
+/// List of HTML tags allowed in the Matrix specification, without the rich reply fallback tag.
+pub const ALLOWED_TAGS_STRICT_WITHOUT_REPLY: [&str; 38] = [
```

Comment:
nit: For purposes of public API, the naming template of these feel a bit disorganized. However, this is just a note, and not a concern, this is fine.

Reply:
What names would you suggest?

Reply:
At the moment i can't come up with much, a suggestion would be to strictly put the `STRICT` and `COMPAT` as prefixes (`STRICT_ALLOWED_TAGS`, `COMPAT_ALLOWED_TAGS`), but then that becomes a matter of opinion.

Seeing how this is only just realistically part of internal API for now, i think this is fine for now, maybe later on i'll take a look at this and see that it needs some redoing.